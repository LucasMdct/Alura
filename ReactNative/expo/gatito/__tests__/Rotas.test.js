import React from "react";
import { render } from "@testing-library/react-native";
import { View } from 'react-native';
import Rotas from '../src/Rotas';

import Servicos from "../src/telas/Servicos";
import Carrinho from "../src/telas/Carrinho";

jest.mock('../src/telas/Servicos');
jest.mock('../src/telas/Carrinho');

describe('Rotas', () => {
    it('Deve renderizar as rotas', () => {
        // Configuração do mock de navegação
        Servicos.mockImplementation(() => <View testID="Servicos" />);

        Carrinho.mockImplementation(() => <View testID="Carrinho" />);

        const { toJSON  } = render(<Rotas />);

        expect(toJSON()).toMatchSnapshot();

    });
});