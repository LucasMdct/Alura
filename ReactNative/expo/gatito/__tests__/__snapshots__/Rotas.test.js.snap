// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rotas Deve renderizar as rotas 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onGestureCancel={[Function]}
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          testID="Servicos"
        />
      </View>
    </RNSScreen>
  </RNSScreenContainer>
  <View
    collapsable={false}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 70,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Servicos, tab, 1 of 2"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "#A050BE",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#A050BE",
              },
              {
                "fontSize": 10,
              },
              {
                "backgroundColor": "#FABE50",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 21,
                "marginTop": 3,
                "paddingTop": 21,
                "width": "100%",
              },
            ]
          }
        >
          Servicos
        </Text>
      </View>
      <View
        accessibilityLabel="Carrinho, tab, 2 of 2"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "#FABE50",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#fff",
              },
              {
                "fontSize": 10,
              },
              {
                "backgroundColor": "#FABE50",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 21,
                "marginTop": 3,
                "paddingTop": 21,
                "width": "100%",
              },
            ]
          }
        >
          Carrinho
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
